### DISCUSSION TITLE
V7.0 Beta27

### DISCUSSION THREAD

**Discussion by subframe7536** [subframe7536/maple-font#255] [General]
- add comma to escape class
- more ignore cases in ligatures
- optimize `>=` / `<=`
- add `=<=` / `=>=`
- add percent in cv01 #245

The ligatures will be changed in next release. Here is the proposal:

### Character Variant (cvXX)

- cv01: `@ $ & % Q => ->` without gap
- cv02: `a` with top arm
- cv03: `i` without left bottom bar
- cv04: `l` with left bottom bar, like consolas

#### italic
- cv31: italic `a` with top arm
- cv32: italic `f` without bottom tail
- cv33: italic `i j` with left bottom bar and horizen top bar
- cv34: italic `k` without circle
- cv35: italic `l` without tail
- cv36: italic `x` without tail

### Style Sets (ssXX)

- ss01: disable equals
- ss02: disable `<= >=`
- ss03: enable arbitrary tag
- ss04: disable `__ #__`
- ss05: disable escape in letter `\n \r \t \b \f`
- ss06: disable `al / ul / il / ff` ...
- ss07: disable `<< >> <<< >>>`

<hr /><em>This discussion was created from the release <a href='https://github.com/subframe7536/maple-font/releases/tag/v7.0-beta27'>V7.0 Beta27 </a>.</em>

**Comment by subframe7536** [comment]
Escape letters like `\n \r \b \t \w` will not enabled by default

**Comment by subframe7536** [comment]
Add alternative `l` in cv36, `x` move to `cv37`

![recording](https://github.com/user-attachments/assets/e4bdb8e5-21db-49c1-943a-bdbd848b9a7f)

**Reply by subframe7536** [reply]
`cv04` should be reused

**Comment by subframe7536** [comment]
3 italic style of `l` (`cv04`, `cv35`, `ss06`) with `i`
![recording](https://github.com/user-attachments/assets/47cf059e-47c4-4550-84e2-82a64e7d90ac)

**Reply by subframe7536** [reply]
`ss06` should not override `cv04` or `cv35`

**Comment by subframe7536** [comment]
By default:
- `<<` `<<<` should have tailing space
- `>>` `>>>` should have leading space

use `ss07` to loose

**Comment by subframe7536** [comment]
Final version ?
![recording](https://github.com/user-attachments/assets/c51eea77-60ee-4549-ab65-7ad2a0b62550)

**Comment by matthew-e-brown** [comment]
Wow, this is turning into a very versatile font! Awesome :)

**Comment by subframe7536** [comment]
There seems no existing tool to easily change the width and properly handling the transformation of glyphs' stems and bars at same time🤔

Especially for composite glyphs and ligatures.

**Comment by subframe7536** [comment]
Escape ligatures for letters like `\w \n` should not enable by default

Will use `ss05` to enable it
